pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -70,6 +71,21 @@ def _move_float_tensors_to_half(self, batch: Any):
         return batch
 
 
+@TrainingTypePluginsRegistry.register("deepspeed", description="Default DeepSpeed Plugin")
+@TrainingTypePluginsRegistry.register("deepspeed_stage_2", description="DeepSpeed with ZeRO Stage 2 enabled", stage=2)
+@TrainingTypePluginsRegistry.register(
+    "deepspeed_stage_2_offload",
+    description="DeepSpeed with ZeRO Stage 2 enabled and Offload",
+    stage=2,
+    cpu_offload=True
+)
+@TrainingTypePluginsRegistry.register("deepspeed_stage_3", description="DeepSpeed with ZeRO Stage 3 enabled", stage=3)
+@TrainingTypePluginsRegistry.register(
+    "deepspeed_stage_3_offload",
+    description="DeepSpeed with ZeRO Stage 3 enabled and Offload",
+    stage=3,
+    cpu_offload=True
+)
```

Comment:
Amazing!!! Great job @kaushikb11 :)

I'm just a bit worried that this looks a bit unwieldy; a lot of decorators like this might make the code ugly, but I'm actually not sure. Any thoughts here?

Maybe we can also offer a function that the plugin can override (like `cls.register_plugins(...)`) that we register instead:

```python
def register_plugins(plugin_registry):
    plugin_register.register(...)
    plugin_register.register(...)
    plugin_register.register(...)
    ...
```

Reply:
Great idea! Added :)